zha/application/platforms/sensor/__init__.py

```
@@ -686,23 +683,17 @@ def __init__(
         super().__init__(cluster_handlers, endpoint, device, **kwargs)
         self._attr_extra_state_attribute_names: set[str] = {
             "measurement_type",
-            self._max_attribute_name,
+            self._attr_max_attribute_name,  # None values are ignored in HA
         }
 
-    @property
-    def _max_attribute_name(self) -> str:
-        """Return the max attribute name."""
-        return self._attr_max_attribute_name or f"{self._attribute_name}_max"
-
     @property
     def state(self) -> dict[str, Any]:
         """Return the state for this sensor."""
         response = super().state
         if self._cluster_handler.measurement_type is not None:
             response["measurement_type"] = self._cluster_handler.measurement_type
 
-        max_attr_name = self._max_attribute_name
-        if not hasattr(self._cluster_handler.cluster.AttributeDefs, max_attr_name):
```

Explanation:
Previously, `_max_attribute_name` returned the main sensor attribute name with a `_max` suffix if no explicit attribute name was provided, so we needed to check if the attribute even exists.

Now, we need to explicitly set the `_attr_max_attribute_name` for each sensor if we want to include an extra state attribute, hence we don't need to check if the attribute actually exists anymore.

(If we still manage to set `_attr_max_attribute_name` to an invalid attribute name, we probably want it to error out, instead of silently failing and not including the extra state attribute.)